### PR TITLE
CAV-198 - Added open API documentation

### PR DESCRIPTION
**Pull request by brdgeth**
Added Open API documentation for the cip phone number API - 3 endpoints.

### CODE REVIEW

File: conf/prod.routes [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Add all the application routes to the app.routes file
 ->         /customer-insight-platform/phone-number              app.Routes
 ->         /                                                    health.Routes
+->         /                                                    definition.Routes
```
Comment (johno1985) [inline conf/prod.routes:4]: would this conflict with the health routes?

File: resources/public/api/conf/1.0/application.yaml [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+openapi: "3.0.3"
+
+info:
+  version: 1.0.0
+  title: CIP phone number API
+  description: |
+    # Overview
+    This is an API that allows an phone number to be verified by calling the Verify endpoint followed by the Verify Passcode endpoint.  It also provides a status API to check the status of a previous Verify request.
+servers:
+  - url: https://test-api.service.hmrc.gov.uk/customer-insight-platform/phone-number
```
Comment (johno1985) [inline resources/public/api/conf/1.0/application.yaml:11]: can we use the MDTP urls? These ones are for API Platform

**Review by johno1985 — COMMENTED** [review]
minor comments

File: conf/prod.routes (hunk 1 above)
Comment (brdgeth) [reply conf/prod.routes]: I don't think so, but I have moved the /api to this file just incase

File: resources/public/api/conf/1.0/application.yaml (hunk 2 above)
Comment (brdgeth) [reply resources/public/api/conf/1.0/application.yaml]: I have added the correct urls now

File: resources/public/api/conf/1.0/application.yaml [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+
+info:
+  version: 1.0.0
+  title: CIP phone number API
+  description: |
+    # Overview
+    This is an API that allows an phone number to be verified by calling the Verify endpoint followed by the Verify Passcode endpoint.  It also provides a status API to check the status of a previous Verify request.
+
+servers:
+  - url: https://www.qa.tax.service.gov.uk/customer-insight-platform/phone-number
```
Comment (johno1985) [inline resources/public/api/conf/1.0/application.yaml:12]: They should all be the same 'https://cip-phone-number-verification.protected.mdtp/customer-insight-platform/phone-number'. The protected microservices have the same hostname in all environments

Comment (brdgeth) [reply resources/public/api/conf/1.0/application.yaml]: fixed

**Review by johno1985 — APPROVED** [review]

**Pull request merged**